### PR TITLE
If GrGLContext setup fails, log OpenGL platform specific information.

### PR DESCRIPTION
**Pull request by chinmaygarde**
Something like the following will be dumped to the logs. The dumps should work on both desktop and mobile GL setups.

```
          Vendor: Apple Inc.
        Renderer: Apple A8 GPU
         Version: 2.0 Apple ES
  Shader Version: 1.0 ES
      Extensions: 43 Available
                  GL_APPLE_clip_distance
                  GL_APPLE_color_buffer_packed_float
                  GL_APPLE_copy_texture_levels
                  GL_APPLE_framebuffer_multisample
                  GL_APPLE_rgb_422
                  GL_APPLE_sync
                  GL_APPLE_texture_format_BGRA8888
                  GL_APPLE_texture_max_level
                  GL_APPLE_texture_packed_float
                  GL_EXT_blend_minmax
                  GL_EXT_color_buffer_half_float
                  GL_EXT_debug_label
                  GL_EXT_debug_marker
                  GL_EXT_discard_framebuffer
                  GL_EXT_draw_instanced
                  GL_EXT_instanced_arrays
                  GL_EXT_map_buffer_range
                  GL_EXT_occlusion_query_boolean
                  GL_EXT_pvrtc_sRGB
                  GL_EXT_read_format_bgra
                  GL_EXT_sRGB
                  GL_EXT_separate_shader_objects
                  GL_EXT_shader_framebuffer_fetch
                  GL_EXT_shader_texture_lod
                  GL_EXT_shadow_samplers
                  GL_EXT_texture_filter_anisotropic
                  GL_EXT_texture_rg
                  GL_EXT_texture_storage
                  GL_IMG_read_format
                  GL_IMG_texture_compression_pvrtc
                  GL_OES_depth24
                  GL_OES_depth_texture
                  GL_OES_depth_texture_cube_map
                  GL_OES_element_index_uint
                  GL_OES_fbo_render_mipmap
                  GL_OES_mapbuffer
                  GL_OES_packed_depth_stencil
                  GL_OES_rgb8_rgba8
                  GL_OES_standard_derivatives
                  GL_OES_texture_float
                  GL_OES_texture_half_float
                  GL_OES_texture_half_float_linear
                  GL_OES_vertex_array_object
```

### CODE REVIEW

**Comment by eseidelGoogle** [comment]
This is *amazing* thank you. :)

File: sky/shell/gpu/direct/ganesh_canvas.cc [hunk 1]
```diff
@@ -32,6 +33,22 @@ bool GaneshCanvas::SetupGrGLInterface() {
       kOpenGL_GrBackend,
       reinterpret_cast<GrBackendContext>(GrGLCreateNativeInterface())));
 
+#ifdef NDEBUG
```
Comment (abarth) [inline sky/shell/gpu/direct/ganesh_canvas.cc]: I'd flip this and put ifndef NDEBUG first, just because that's the more common pattern.

**Comment by chinmaygarde** [comment]
Fixes https://github.com/flutter/flutter/issues/5171. Needed the extensions stuff so I can pick the correct fencing routines for an unrelated patch.

**Comment by abarth** [comment]
LGTM

File: sky/shell/gpu/direct/ganesh_canvas.cc (hunk 1 above)
Comment (chinmaygarde) [inline sky/shell/gpu/direct/ganesh_canvas.cc]: Done.

**Pull request merged**